src/replication.c

```diff
@@ -2167,13 +2173,20 @@ void syncWithMaster(connection *conn) {
     /* AUTH with the master if required. */
     if (server.repl_state == REPL_STATE_SEND_AUTH) {
         if (server.masteruser && server.masterauth) {
-            err = sendSynchronousCommand(SYNC_CMD_WRITE,conn,"AUTH",
-                                         server.masteruser,server.masterauth,NULL);
+            sds masteruser = sdsnew(server.masteruser);
+            sds auth = sdsnew("AUTH");
+            err = sendSynchronousCommand(SYNC_CMD_WRITE | SYNC_CMD_WRITE_SDS, conn, auth,
+                                         masteruser, server.masterauth, NULL);
+            sdsfree(masteruser);
+            sdsfree(auth);
             if (err) goto write_error;
             server.repl_state = REPL_STATE_RECEIVE_AUTH;
             return;
         } else if (server.masterauth) {
-            err = sendSynchronousCommand(SYNC_CMD_WRITE,conn,"AUTH",server.masterauth,NULL);
+            sds auth = sdsnew("AUTH");
+            err = sendSynchronousCommand(SYNC_CMD_WRITE | SYNC_CMD_WRITE_SDS, conn, auth,
+                    server.masterauth, NULL);
+            sdsfree(auth);
             if (err) goto write_error;
             server.repl_state = REPL_STATE_RECEIVE_AUTH;
             return;
```

Comment:
Seems like these 2 cases could be combined to eliminate the duplicate code.  Something like:
```C
sds masteruser = (server.masteruser) ? sdsnew(server.masteruser) : NULL;
...
if (masteruser) err = sendSynchronousCommand(... with masteruser ...)
else            err = sendSynchronousCommand(... without masteruser ...)
...
sdsfree(masteruser);
```

Reply:
Re-factored this part of the code a bit to remove duplicates